tests/unit/dcc_comms/test_mqtt_dcc_comms.py

```diff
@@ -0,0 +1,193 @@
+        :return:
+        """
+        # Create Edge System
+        self.edge_system = Dell5KEdgeSystem("TestGateway")
+
+        self.url = "Broker-IP"
+        self.port = "Broker-Port"
+        self.clean_session = True
+        self.publish_message = "test-message"
+
+        # Creating messaging attributes
+        self.mqtt_msg_attr = MqttMessagingAttributes(pub_topic="publish-topic")
+
+        # Instantiate client for DCC communication
+        self.client = MqttDccComms(edge_system_name=self.edge_system.name,
+                                   url=self.url, port=self.port, clean_session=self.clean_session)
+
+    def tearDown(self):
+        """
+        Clean up all parameters used in test cases
+        :return:
+        """
+        self.edge_system = None
+        self.url = None
+        self.port = None
+        self.clean_session = None
+        self.publish_message = None
+        self.mqtt_msg_attr = None
+        self.client = None
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_no_msgattr_no_client_id(self, mock_connect):
+        """
+        Test MqttDccComms initialisation with empty client ID
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+        MqttDccComms(edge_system_name=self.edge_system.name,
+                     url=self.url, port=self.port,
+                     clean_session=self.clean_session)
+
+        mock_connect.assert_called()
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_no_msgattr_with_client_id(self, mock_connect):
+        """
+        Test MqttDccComms initialisation with non-empty client ID
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+        MqttDccComms(edge_system_name=self.edge_system.name,
+                     url=self.url, port=self.port,
+                     client_id="unit-test", clean_session=self.clean_session)
+
+        mock_connect.assert_called()
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_with_valid_msgattr(self, mock_connect):
+        """
+        Test MqttDccComms initialisation with valid messaging attributes
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+
+        # Test with valid messaging attributes
+        MqttDccComms(edge_system_name=self.edge_system.name,
+                     url=self.url, port=self.port, clean_session=self.clean_session,
+                     mqtt_msg_attr=self.mqtt_msg_attr)
+
+        mock_connect.assert_called()
+
+    def test_init_with_invalid_msgattr(self):
+        """
+        Test initialisation with invalid messaging attributes
+        :return:
+        """
+        self.assertRaises(TypeError, lambda: MqttDccComms(edge_system_name=self.edge_system.name,
+                                                          url=self.url, port=self.port,
+                                                          clean_session=self.clean_session, mqtt_msg_attr=""))
+
```

Comment:
Test case for _connect() ?

Mock transport's init() and check for all attributes using assertcalled with